app/controllers/bulk_submission_forms_controller.rb

```diff
@@ -18,7 +18,12 @@ def edit
   end
 
   def update
-    @form = BulkSubmissionForm.new(bulk_submission_form_params.merge(bulk_submission: BulkSubmission.find(params[:id])))
+    @form = BulkSubmissionForm.new(
+              bulk_submission_form_params.merge(
+                bulk_submission: BulkSubmission.find(params[:id]),
```

Comment:
Isn't calling params direct a no-no

Shouldn't there be a 
```ruby
def submission_id_param
    params.permit(:id)
end
```

There seems to be a lot of references to params in the rest of the controller... maybe a refactor is on order?

Reply:
this always gives me a headache. 

1. As soon as i allow list :id, as above, I get errors 
```
unknown attribute 'id' for BulkSubmissionForm
...
{"_method"=>"patch", "authenticity_token"=>"[FILTERED]", "commit"=>"continue", "id"=>"88898b08-80a5-46c4-921b-dee3f0f9b305"}
```
 _i think because BulkSubmissionForm is a fake model (i,e, form) and so does not have an id and rails is doing some magic?!_ 
2. I think that if you are not using the params (params[:id]) for database record creation it does not matter as it cannot be used for mass assignment (as in `BulkSubmission.create(params)`)

**Happy to chat about this as it may resolve my own [lack of] understanding of how "Strong params" work**

Reply:
it works when i add a separate strong params method

```
  def bulk_submission_id_params
    params.permit(:id)
  end

  def bulk_submission_form_params
    params.permit(:uploaded_file)
  end
```
...
but not with one
```
  def bulk_submission_form_params
    params.permit(:id, :uploaded_file)
  end
```

do you know why this is?

Reply:
Discussed and happy for this to be merged and the params options explored in a separate branch :+1: